lighthouse-core/audits/redirects.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit');
+const Util = require('../report/v2/renderer/util');
+const UnusedBytes = require('./byte-efficiency/byte-efficiency-audit');
+
+// PSI allows one redirect (http://example.com => http://m.example.com)
+const REDIRECT_THRESHOLD = 1;
+
+class Redirects extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'redirects',
+      description: 'Avoids page redirects.',
+      failureDescription: 'Has page redirects.',
+      helpText: ' Redirects introduce additional delays before the page can be loaded. [Learn more](https://developers.google.com/speed/docs/insights/AvoidRedirects).',
+      requiredArtifacts: ['URL', 'devtoolsLogs'],
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    return artifacts.requestMainResource(artifacts.devtoolsLogs[Audit.DEFAULT_PASS])
+      .then(mainResource => {
+        // redirects is only available when redirects happens
+        const redirectRequests = mainResource.redirects || [];
+        let totalWastedMs = 0;
+        let debugString = null;
+
+        const pageRedirects = redirectRequests.map(request => {
+          const wastedMs = (request.endTime - request.startTime) * 1000;
```

Comment:
I don't think this is what we want here, from the screenshot of gmail it looks like `request.endTime` is the end time of the final mainResource, I think we want something more like `requests[1].startTime - requests[0].startTime`?